OpenRA.Mods.Common/Projectiles/LaserZap.cs

```diff
@@ -126,9 +129,12 @@ public LaserZap(LaserZapInfo info, ProjectileArgs args, Color color)
 
 		public void Tick(World world)
 		{
+			// Position of firing actor might have changed
+			source = args.Source;
```

Comment:
Actually, this doesn't change. Looks like you want to use `args.CurrentSource()` (see `AreaBeam`).